plugins/rss.ts

```diff
@@ -0,0 +1,90 @@
+import { $XML, stringify } from "../deps/xml.ts";
+import { Page } from "../core/filesystem.ts";
+import { Search } from "../plugins/search.ts";
+
+import type { Data, Site } from "../core.ts";
+
+export interface Options {
+  filename: string;
+  query: string;
+  sort: string;
+  link: string;
+  title: string;
+  buildDate: Date;
+  description: string;
+  language: string;
+}
+
+export const defaults: Options = {
+  filename: "/feed.rss",
+  query: "",
+  sort: "url=asc",
+  link: "http://127.0.0.1:3000",
+  title: "My RSS Feed",
+  buildDate: new Date(),
+  description: "",
+  language: "en",
+};
+
+export default (userOptions?: Partial<Options>) => {
+  const options = {
+    ...defaults,
+    ...userOptions,
+  };
+
+  return (site: Site) => {
+    site.addEventListener("afterRender", () => {
+      const feed = Page.create(options.filename, getFeedContent(site));
+      site.pages.push(feed);
+    });
+
+    const getFeedContent = (site: Site) => {
+      const search = new Search(site, true);
+      const pages = search.pages(options.query, options.sort);
+      const items = pages.map((page: Data) => ({
+        title: page.title,
+        link: `${options.link}${page.url}`,
+        guid: {
+          "@isPermaLink": false,
+          "#text": `${options.link}${page.url}`,
+        },
+        description: page.excerpt,
+        "content:encoded": {
+          "#text": page.content,
```

Comment:
`page.content` returns the full HTML code, [maybe `page.data.children` fit better here](https://lume.land/blog/posts/lume-1.16.0-release-notes/#page.data.children-property).

```suggestion
          "#text": page.data.children.toString(),
```

Reply:
I don't think I can access `children` in this scope